lib/ecto/query/builder/cte.ex

```diff
@@ -35,9 +35,9 @@ defmodule Ecto.Query.Builder.CTE do
   If possible, it does all calculations at compile time to avoid
   runtime work.
   """
-  @spec build(Macro.t, Macro.t, Macro.t, Macro.Env.t) :: Macro.t
-  def build(query, name, cte, env) do
-    Builder.apply_query(query, __MODULE__, [escape(name, env), build_cte(name, cte, env)], env)
+  @spec build(Macro.t, Macro.t, Macro.t, boolean(), Macro.Env.t) :: Macro.t
+  def build(query, name, cte, materialized, env) do
+    Builder.apply_query(query, __MODULE__, [escape(name, env), build_cte(name, cte, env), materialized], env)
```

Comment:
```suggestion
  def build(query, name, cte, materialized?, env) do
    Builder.apply_query(query, __MODULE__, [escape(name, env), build_cte(name, cte, env), materialized?], env)
```

I think the convention is to add a trailing question mark  but not 100% sure. I'm thinking of things like `changeset.valid?`. cc @josevalim